src/masternodes/poolpairs.h

```diff
@@ -131,7 +131,7 @@ class CPoolPair : public CPoolPairMessage
             CAmount liqB = (arith_uint256(amountB) * arith_uint256(totalLiquidity) / reserveB).GetLow64();
             liquidity = std::min(liqA, liqB);
 
-            if ((std::max(liqA, liqB) - liquidity) / liquidity * 100 > 3) {
+            if ((std::max(liqA, liqB) - liquidity) / liquidity * 100 >= 3) {
```

Comment:
Checking liquidity that equals to 0 should be before divide by potentially 0.

Reply:
liquidity == 0 is done at line 122. This has excluded liquidity == 0's case.

Reply:
Line 122 is `totalLiquidity` here is exactly where it should be checked. Line above `liquidity` is calculated as min of 2 liquidities and it can be 0.

Reply:
Can you submit a fix please? @bvbfan

Reply:
Nope it cannot be 0. But I'm okay with a safety net fix here 


```suggestion
            if (liquidity > 0 && (std::max(liqA, liqB) - liquidity) / liquidity * 100 >= 3) {
```

Reply:
`(std::max(liqA, liqB) - liquidity) * 100 / liquidity` should be OK